File: backend/Shared/DB/Entities/StudentEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace VisaCenterBackend.Shared.DB.Entities;

/// Студент
public class StudentEntity : AppBaseEntity {
  /// Пользователь к которому привязан сотрудник
  [ForeignKey(nameof(UserId))]
  public UserEntity User { get; set; } = null!;
  [Key] public Guid UserId { get; set; }

  /// Уведомление о прибытии студента
  [ForeignKey(nameof(ArrivalNoticeId))]
  public StudentArrivalNoticeEntity? ArrivalNotice { get; set; }
  public Guid ArrivalNoticeId { get; set; }

  /// Миграционная карта студента
  [ForeignKey(nameof(MigrationCardId))]
  public StudentMigrationCardEntity? MigrationCard { get; set; }
  public Guid MigrationCardId { get; set; }

  /// Виза студента. Её id - строка, идентификатор самой визы
  [ForeignKey(nameof(VisaId))]
  public StudentVisaEntity? Visa { get; set; }
  public string VisaId { get; set; } = null!;

  /// Паспорт студента
  [ForeignKey(nameof(PassportId))]
  public StudentPassportEntity? Passport { get; set; }
  public Guid PassportId { get; set; }

  /// Телефон
  public string? Phone { get; set; }

  /// Куратор
  public string? Curator { get; set; }

  /// Факультет
  public string? Faculty { get; set; }

  /// Курс
  public int? Course { get; set; }

  /// Группа
  public string? Group { get; set; }

  /// Близкие родственники
  public List<StudentCloseRelativeEntity> CloseRelatives { get; set; } = null!;

  /// Визовые анкеты
  public List<StudentVisaRequestEntity> VisaRequests { get; set; } = null!;
}

public class StudentEntityConfiguration : IEntityTypeConfiguration<StudentEntity> {
  public void Configure(EntityTypeBuilder<StudentEntity> builder) {
    builder.ToTable(nameof(StudentEntity));
    builder.HasQueryFilter(e => e.DeletedAt == null);
  }
}

File: backend_nest/src/api/student/student.entity.ts
// import BaseEntity from '@common/BaseEntity/base-entity';
// import {
//   Field, ID, Int, ObjectType,
// } from '@nestjs/graphql';
// import { Student } from '@prisma-client';
// import { OptionalNullable } from '../../type-utils';
// import UserEntity from '../user/user.entity';
// import StudentArrivalNoticeEntity from './student-arrival-notice/student-arrival-notice.entity';
// import StudentMigrationCardEntity from './student-migration-card/student-migration-card.entity';
// import StudentPassportEntity from './student-passport/student-passport.entity';
// import StudentVisaEntity from './student-visa/student-visa.entity';
//
// // Студент
// @ObjectType({ description: 'Студент' })
// export default class StudentEntity extends BaseEntity implements OptionalNullable<Student> {
//   // id Пользователя
//   @Field(() => ID, { description: 'id Пользователя' })
//   id: string;
//
//   // Пользователь
//   @Field(() => UserEntity, { description: 'Пользователь' })
//   user: UserEntity;
//
//   // Уведомление о прибытии
//   @Field(() => StudentArrivalNoticeEntity, { nullable: true, description: 'Уведомление о прибытии' })
//   arrivalNotice?: StudentArrivalNoticeEntity | null;
//
//   // Миграционная карта
//   @Field(() => StudentMigrationCardEntity, { nullable: true, description: 'Миграционная карта' })
//   migrationCard?: StudentMigrationCardEntity | null;
//
//   // Виза
//   @Field(() => StudentVisaEntity, { nullable: true, description: 'Виза' })
//   visa?: StudentVisaEntity | null;
//
//   // Паспорт
//   @Field(() => StudentPassportEntity, { nullable: true, description: 'Паспорт' })
//   passport?: StudentPassportEntity | null;
//
//   // Телефон
//   @Field(() => String, { nullable: true, description: 'Телефон' })
//   phone?: string | null;
//
//   // Куратор
//   @Field(() => String, { nullable: true, description: 'Куратор' })
//   curator?: string | null;
//
//   // Факультет
//   @Field(() => String, { nullable: true, description: 'Факультет' })
//   faculty?: string | null;
//
//   // Курс
//   @Field(() => Int, { nullable: true, description: 'Курс' })
//   course?: number | null;
//
//   // Группа
//   @Field(() => String, { nullable: true, description: 'Группа' })
//   group?: string | null;
// }
